src/uu/tr/src/operation.rs

```diff
@@ -0,0 +1,529 @@
+                        (start..=end)
+                            .map(|c| std::char::from_u32(c).unwrap())
+                            .collect(),
+                    )
+                } else {
+                    Sequence::CharRange((start..=end).filter_map(std::char::from_u32).collect())
+                }
+            })
+        })
+    }
+
+    fn parse_char_star(input: &str) -> IResult<&str, Sequence> {
+        tuple((tag("["), take(1usize), tag("*"), tag("]")))(input).map(|(_, (_, _, _, _))| todo!())
+    }
+
+    fn parse_char_repeat(input: &str) -> IResult<&str, Sequence> {
+        tuple((tag("["), take(1usize), tag("*"), take_until("]"), tag("]")))(input).map(
+            |(l, (_, c, _, n, _))| {
+                (
+                    l,
+                    Sequence::CharRange(
+                        std::iter::repeat(c.chars().next().unwrap())
+                            .take(n.parse().unwrap())
+                            .collect(),
+                    ),
+                )
+            },
+        )
+    }
+
+    fn parse_alnum(input: &str) -> IResult<&str, Sequence> {
+        tag("[:alnum:]")(input).map(|(l, _)| {
+            (
+                l,
+                Sequence::CharRange(('a'..='z').chain('A'..'Z').chain('0'..'9').collect()),
+            )
+        })
+    }
+
+    fn parse_alpha(input: &str) -> IResult<&str, Sequence> {
+        tag("[:alpha:]")(input).map(|(l, _)| {
+            (
+                l,
+                Sequence::CharRange(('a'..='z').chain('A'..'Z').collect()),
+            )
+        })
+    }
+
+    fn parse_blank(input: &str) -> IResult<&str, Sequence> {
+        tag("[:blank:]")(input).map(|(_, _)| todo!())
+    }
+
+    fn parse_control(input: &str) -> IResult<&str, Sequence> {
+        tag("[:cntrl:]")(input).map(|(_, _)| todo!())
+    }
+
+    fn parse_digit(input: &str) -> IResult<&str, Sequence> {
+        tag("[:digit:]")(input).map(|(l, _)| (l, Sequence::CharRange(('0'..='9').collect())))
+    }
+
+    fn parse_graph(input: &str) -> IResult<&str, Sequence> {
+        tag("[:graph:]")(input).map(|(_, _)| todo!())
+    }
+
+    fn parse_lower(input: &str) -> IResult<&str, Sequence> {
+        tag("[:lower:]")(input).map(|(l, _)| (l, Sequence::CharRange(('a'..='z').collect())))
+    }
+
+    fn parse_print(input: &str) -> IResult<&str, Sequence> {
+        tag("[:print:]")(input).map(|(_, _)| todo!())
+    }
+
+    fn parse_punct(input: &str) -> IResult<&str, Sequence> {
+        tag("[:punct:]")(input).map(|(_, _)| todo!())
+    }
+
+    fn parse_space(input: &str) -> IResult<&str, Sequence> {
+        tag("[:space:]")(input).map(|(_, _)| todo!())
+    }
+
+    fn parse_upper(input: &str) -> IResult<&str, Sequence> {
+        tag("[:upper:]")(input).map(|(l, _)| (l, Sequence::CharRange(('A'..='Z').collect())))
+    }
+
+    fn parse_xdigit(input: &str) -> IResult<&str, Sequence> {
+        tag("[:xdigit:]")(input).map(|(_, _)| todo!())
```

Comment:
This should be fairly easy to implement. From some quick tests I think it should be `('0'..='9').chain('A'..='Z').chain('a'..='z')`

Here's what I tested:
```shell
❯ echo "0123456789ABCDEFabcdef" | tr "[:xdigit:]" "abcdefghijklmnopqrstuvwxyz"
abcdefghijklmnopqrstuv

❯ echo "0123456789abcdefABCDEF" | tr "[:xdigit:]" "abcdefghijklmnopqrstuvwxyz"
abcdefghijqrstuvklmnop
```

Reply:
From the [grep manual](https://www.gnu.org/software/grep/manual/html_node/Character-Classes-and-Bracket-Expressions.html). `xdigit` is simply hexadecimal digits and I have implemented it as such.